homeassistant/components/hdmi_cec/media_player.py

```diff
@@ -145,25 +145,26 @@ def volume_down(self) -> None:
         _LOGGER.debug("%s: volume down", self._logical_address)
         self.send_keypress(KEY_VOLUME_DOWN)
 
-    @property
-    def state(self) -> str | None:
-        """Cache state of device."""
-        return self._state
+    def _hdmi_cec_unavailable(self, callback_event):
+        # Change state to unknown. Without this, entity would remain in
+        # its last state, since the state changes are pushed.
+        self._attr_state = None
```

Comment:
This is not the same. You should set `self._attr_unavailable = True`. It should be set to `False` inside `update`

Reply:
There is a test specifically to prevent the behavior you ask for, so I didn't implement that.

Reply:
Hey @epenet,

The tests were setup to test the current (not always correct) behavior & to allow updating the code from there. I think it'd be good to change it as suggested and update the tests. I'm happy to help if that's needed.

Reply:
I've adjusted the test accordingly